qemu/tests/block_hotplug.py

```diff
@@ -87,17 +87,23 @@ def get_new_disk(disk1, disk2):
         for num in xrange(blk_num):
             device = qdevices.QDevice(pci_type)
             if params.get("need_plug") == "yes":
+                disks_before_plug = find_disk(vm, get_disk_cmd)
+
                 if params.get("need_controller", "no") == "yes":
                     controller_model = params.get("controller_model")
                     controller = qdevices.QDevice(controller_model)
+                    bus_extra_param = params.get("bus_extra_params_%s" % img_list[num + 1])
+                    if bus_extra_param and "iothread" in bus_extra_param:
+                        match = re.search("iothread=(\w+)", bus_extra_param)
+                        if match:
+                            qdevice_params = {"iothread": match.group(1)}
+                            controller = qdevices.QDevice(controller_model, params=qdevice_params)
```

Comment:
I am curious about:

1. why not update the old object instead of creating a new one?
2. will a new `iothread` be plugged at this time?

Reply:
@luckyh, have updated, thanks for pointing it out.